src/utils/auth.ts

```diff
@@ -0,0 +1,40 @@
+import express, {Request, NextFunction} from 'express';
+const jwt = require('jsonwebtoken');
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
+
+export interface ReqWithUser extends Request {
+    user: {
+        id: number;
+        email: string;
+        dateCreated: Date;
+        username: string;
+        password?: string;
+    }
+}
+
+// TODO: Test out this middleware once login endpoint is done and returns a JSON web token
+const authenticateJWT = async (req: ReqWithUser, next: NextFunction, prisma: PrismaClient) => {
+    const authHeader = req.headers.authorization;
+
+    if (authHeader) {
```

Comment:
Let's also double-check it propertly starts with the prefix
```suggestion
    if (authHeader && authHeader.startsWith("Bearer ")) {
```